pkg/cmd/workflow/run/run.go

```diff
@@ -0,0 +1,382 @@
+				providedInputs[inputName] = providedValue
+			}
+		}
+	}
+
+	for inputName, inputValue := range inputs {
+		if inputValue.Required && providedInputs[inputName] == "" {
+			return fmt.Errorf("missing required input '%s'", inputName)
+		}
+	}
+
+	path := fmt.Sprintf("repos/%s/actions/workflows/%d/dispatches",
+		ghrepo.FullName(repo), workflow.ID)
+
+	requestByte, err := json.Marshal(struct {
+		Ref    string            `json:"ref"`
+		Inputs map[string]string `json:"inputs"`
+	}{
+		Ref:    ref,
+		Inputs: providedInputs,
+	})
+	if err != nil {
+		return fmt.Errorf("failed to serialize workflow inputs: %w", err)
+	}
+
+	body := bytes.NewReader(requestByte)
+
+	err = client.REST(repo.RepoHost(), "POST", path, body, nil)
+	if err != nil {
+		return fmt.Errorf("could not create workflow dispatch event: %w", err)
+	}
+
+	if opts.IO.CanPrompt() {
+		out := opts.IO.Out
+		cs := opts.IO.ColorScheme()
+		fmt.Fprintf(out, "%s Created workflow_dispatch event for %s at %s\n",
+			cs.SuccessIcon(), cs.Cyan(workflow.Base()), cs.Bold(ref))
+
+		fmt.Fprintln(out)
+
+		fmt.Fprintf(out, "To see runs for this workflow, try: %s\n",
+			cs.Boldf("gh run list --workflow=%s", workflow.Base()))
+	}
+
+	return nil
+}
+
+type WorkflowInput struct {
+	Required    bool
+	Default     string
+	Description string
+}
+
+func findInputs(yamlContent []byte) (map[string]WorkflowInput, error) {
+	var rootNode yaml.Node
+	err := yaml.Unmarshal(yamlContent, &rootNode)
+	if err != nil {
+		return nil, fmt.Errorf("unable to parse workflow YAML: %w", err)
+	}
+
+	if len(rootNode.Content) != 1 {
+		return nil, errors.New("invalid yaml file")
+	}
+
+	var onKeyNode *yaml.Node
+	var dispatchKeyNode *yaml.Node
+	var inputsKeyNode *yaml.Node
+	var inputsMapNode *yaml.Node
+
+	// TODO this is pretty hideous
+	for _, node := range rootNode.Content[0].Content {
+		if onKeyNode != nil {
+			for _, node := range node.Content {
+				if dispatchKeyNode != nil {
+					for _, node := range node.Content {
+						if inputsKeyNode != nil {
+							inputsMapNode = node
+							break
+						}
+						if node.Value == "inputs" {
+							inputsKeyNode = node
+						}
+					}
+					break
+				}
+				if node.Value == "workflow_dispatch" {
+					dispatchKeyNode = node
+				}
+			}
+			break
+		}
+		if strings.EqualFold(node.Value, "on") {
+			onKeyNode = node
+		}
+	}
+
+	if onKeyNode == nil {
+		return nil, errors.New("invalid workflow: no 'on' key")
+	}
+
+	if dispatchKeyNode == nil {
+		return nil, errors.New("unable to manually run a workflow without a workflow_dispatch event")
+	}
+
+	out := map[string]WorkflowInput{}
+
+	if inputsKeyNode == nil || inputsMapNode == nil {
+		return out, nil
+	}
+
+	err = inputsMapNode.Decode(&out)
+	if err != nil {
+		return nil, fmt.Errorf("could not decode workflow inputs: %w", err)
+	}
+
+	return out, nil
+}
+
+func getWorkflowContent(client *api.Client, repo ghrepo.Interface, workflow *shared.Workflow, ref string) ([]byte, error) {
```

Comment:
Now that workflow view is merged, which also uses this exact function, we should share it between these commands.